course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
+                scope.searchInProgress = true;
+                expect(scope.disableAddUserButton()).toBe(true);
+            });
+            it('should enable if the search had multiple results and one is selected',
+               function() {
+                   scope.searchResults = [{}]; // contents don't matter, only length
+                   scope.selectedResult = {id: 123};
+                   expect(scope.disableAddUserButton()).toBe(false);
+               }
+            );
+            it('should disable if the search had multiple results and none are selected',
+               function() {
+                   scope.searchResults = [{}]; // contents don't matter, only length
+                   scope.selectedResult = {id: null};
+                   expect(scope.disableAddUserButton()).toBe(true);
+               }
+            );
+            it('should enable if a search term has been entered and there are no results',
+               function() {
+                   scope.searchTerm = 'bob';
+                   expect(scope.disableAddUserButton()).toBe(false);
+               }
+            );
+            it('should disable if there is no search term or results', function() {
+                expect(scope.disableAddUserButton()).toBe(true);
+            });
+        });
+
+        describe('filterResults', function() {
+            // NOTE: relies on compareRoles() working properly.  mocking
+            //       its results wasn't worth it.
+            it('should not blow up on empty input', function() {
+                expect(scope.filterResults([])).toEqual([]);
+            });
+
+            it('should return one role per univ_id', function() {
+                var searchResults = [
+                    {univ_id: 123, active: 0, prime_role_indicator: ''},
+                    {univ_id: 123, active: 1, prime_role_indicator: 'Y'},
+                    {univ_id: 456, active: 0, prime_role_indicator: 'Y'},
+                    {univ_id: 456, active: 1, prime_role_indicator: ''},
+                    {univ_id: 789, active: 0, prime_role_indicator: 'Y'},
+                ];
+                var filtered = scope.filterResults(searchResults);
+                var uniq = {};
+                filtered.forEach(function(r) { uniq[r.univ_id] = true; });
+                var filteredIds = Object.keys(uniq);
+                filteredIds.sort();
+
+                expect(filtered.length).toEqual(3);
+                expect(filteredIds).toEqual(['123', '456', '789']);
+            });
+        });
+
+        describe('handleAjaxError', function() {
+            it('should log an error', function() {
+                var expectedMessage = 
+                    "Error getting data from https://tea.pot: 418 I'm a teapot: {}";
+                scope.handleAjaxError({}, 418, {}, {url: 'https://tea.pot'},
+                                      "I'm a teapot");
+                expect($log.error.logs).toEqual([[expectedMessage]]);
```

Comment:
I *think* it's because `log.error()` et al. are variadic, so even though we only pass in one argument, it gets an array.